datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java

```diff
@@ -1430,4 +1437,55 @@ public FlowIdentifier updateSalt(SdxCluster sdxCluster) {
             return sdxReactorFlowManager.triggerSaltUpdate(sdxCluster);
         }
     }
+
+    public FlowIdentifier horizontalScaleDatalake(String name, StackScaleV4Request scaleRequest) {
+        LOGGER.info("Horizontal scling Data Lake with name {}", name);
+        String accountId = ThreadBasedUserCrnProvider.getAccountId();
+        final SdxCluster sdxCluster = sdxClusterRepository.findByAccountIdAndClusterNameAndDeletedIsNullAndDetachedIsFalse(accountId, name)
+                .orElseThrow(() -> notFound("SDX cluster", name).get());
+        return sdxReactorFlowManager.triggerHorizontalScaleDataLake(sdxCluster, scaleRequest);
+    }
+
+    public void validateHorizontalScaleRequest(SdxCluster sdxCluster, StackScaleV4Request request) {
+        if (!ENTERPRISE.equals(sdxCluster.getClusterShape())) {
+            throw new BadRequestException(String.format("scaling not supported on: %s.", sdxCluster.getClusterShape().name()));
+        }
+        var hostGroup = request.getGroup();
+        if (!InstanceGroupName.valueOf(hostGroup.toUpperCase(Locale.ROOT)).isScalable()) {
+            throw new BadRequestException(String.format("%s host group cannot scale", hostGroup));
+        }
+        DetailedEnvironmentResponse environmentResponse = environmentClientService.getByName(sdxCluster.getEnvName());
+        if (!EnvironmentStatus.upscalable().contains(environmentResponse.getEnvironmentStatus())) {
+            throw new BadRequestException(String.format("The cluster cannot be scale in the %s status.", environmentResponse.getEnvironmentStatus().name()));
+        }
+        StackV4Response stackResponse = stackV4Endpoint.getByCrn(1L, sdxCluster.getStackCrn(), Set.of());
+        if (isNodeRequestInvalid(stackResponse, request)) {
+            throw new BadRequestException("The requested nodeCount hurt the minimum condition of horizontal sclaing.");
+        }
+    }
+
+    private boolean isNodeRequestInvalid(StackV4Response stack, StackScaleV4Request request) {
+        if (request.getDesiredCount() < 0) {
+            LOGGER.warn("Negative nodeCount was requested.");
+            return true;
+        }
+        var hostgroupMap = stack.getInstanceGroups()
+                .stream()
+                .filter(group -> group.getName().equalsIgnoreCase(request.getGroup()))
+                .collect(toMap(InstanceGroupV4Response::getName, InstanceGroupV4Response::getNodeCount));
+        if (hostgroupMap.isEmpty()) {
+            LOGGER.warn("The reqested hostgroup name not found!");
+            return true;
+        }
+        var instanceGroupName = InstanceGroupName.valueOf(request.getGroup().toUpperCase(Locale.ROOT));
+        if (instanceGroupName.getMinimumNodeCount() > request.getDesiredCount()) {
+            LOGGER.warn("Requested nodeCount is less then the minimum nodeCount.");
+            return true;
+        }
```

Comment:
This check should also be performed in the front end. Please update the UI Jira to include this validation as well.